File: converted_source/uprginfo.lfm.java

import java.awt.*;
import java.awt.event.*;
import javax.swing.*;
import javax.swing.event.*;
import javax.swing.table.*;
// import common.*;

public class ProgramInfoWindow extends JFrame {
    private static final long serialVersionUID = 1L;
    SpeedButton helpButton;
    JButton Close;
    JButton saveListAs;
    JButton copyToClipboard;
    JTextArea infoList;
    public JPanel mainContentPane;
    public JPanel delphiPanel;
    
    public ProgramInfoWindow() {
        super();
        this.setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE); // HIDE_ON_CLOSE DO_NOTHING_ON_CLOSE
        initialize();
    }
    
    private void initialize() {
        this.setSize(new Dimension(705, 325));
        this.setContentPane(getMainContentPane());
        this.setTitle("Program Information");
    }
    
    private JPanel getMainContentPane() {
        if (mainContentPane == null) {
            mainContentPane = new JPanel();
            mainContentPane.setLayout(new BorderLayout());
            mainContentPane.add(getDelphiPanel(), BorderLayout.CENTER);
        }
        return mainContentPane;
    }
    public void OnClose(ActionEvent event) {
        System.out.println("Closed");
    }
    
    public JPanel getDelphiPanel() {
        if (delphiPanel != null) return delphiPanel;
        Toolkit toolkit = java.awt.Toolkit.getDefaultToolkit();
        delphiPanel = new JPanel();
        delphiPanel.setLayout(null);
        // -- delphiPanel.setLayout(BoxLayout(contentPane, BoxLayout.Y_AXIS));
        // Font buttonFont = new Font("Arial Narrow", Font.PLAIN, 9);
        // Insets buttonMargin = new Insets(1, 1, 1, 1);
        this.setBounds(310, 266, 450, 301  + 80); // extra for title bar and menu
        //  --------------- UNHANDLED ATTRIBUTE: this.TextHeight = 13;
        //  --------------- UNHANDLED ATTRIBUTE: this.Scaled = False;
        //  --------------- UNHANDLED ATTRIBUTE: this.OnResize = FormResize;
        //  --------------- UNHANDLED ATTRIBUTE: this.PixelsPerInch = 96;
        //  --------------- UNHANDLED ATTRIBUTE: this.AutoScroll = False;
        //  --------------- UNHANDLED ATTRIBUTE: this.Position = poScreenCenter;
        //  --------------- UNHANDLED ATTRIBUTE: this.OnCreate = FormCreate;
        
        
        Image helpButtonImage = toolkit.createImage("../resources/ProgramInfoForm_helpButton.png");
        this.helpButton = new SpeedButton("Help", new ImageIcon(helpButtonImage));
        this.helpButton.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent event) {
                helpButtonClick(event);
            }
        });
        this.helpButton.setMnemonic(KeyEvent.VK_H);
        this.helpButton.setBounds(332, 135, 113, 25);
        
        this.Close = new JButton("Close");
        this.Close.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent event) {
                CloseClick(event);
            }
        });
        this.Close.setMnemonic(KeyEvent.VK_C);
        this.Close.setBounds(332, 4, 113, 21);
        //  --------------- UNHANDLED ATTRIBUTE: this.Close.Cancel = True;
        
        this.saveListAs = new JButton("Save as...");
        this.saveListAs.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent event) {
                saveListAsClick(event);
            }
        });
        this.saveListAs.setMnemonic(KeyEvent.VK_S);
        this.saveListAs.setBounds(332, 83, 113, 21);
        
        this.copyToClipboard = new JButton("Copy to Clipboard");
        this.copyToClipboard.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent event) {
                copyToClipboardClick(event);
            }
        });
        this.copyToClipboard.setMnemonic(KeyEvent.VK_P);
        this.copyToClipboard.setBounds(332, 55, 113, 21);
        //  --------------- UNHANDLED ATTRIBUTE: this.copyToClipboard.Cancel = True;
        
        this.infoList = new JTextArea(10, 10);
        this.infoList.setBounds(2, 2, 323, 295);
        //  --------------- UNHANDLED ATTRIBUTE: this.infoList.ScrollBars = ssVertical;
        
        delphiPanel.add(helpButton);
        delphiPanel.add(Close);
        delphiPanel.add(saveListAs);
        delphiPanel.add(copyToClipboard);
        JScrollPane scroller1 = new JScrollPane();
        scroller1.setBounds(2, 2, 323, 295);;
        scroller1.setViewportView(infoList);
        delphiPanel.add(scroller1);
        return delphiPanel;
    }
    
        
    public void CloseClick(ActionEvent event) {
        System.out.println("CloseClick");
    }
        
    public void copyToClipboardClick(ActionEvent event) {
        System.out.println("copyToClipboardClick");
    }
        
    public void helpButtonClick(ActionEvent event) {
        System.out.println("helpButtonClick");
    }
        
    public void saveListAsClick(ActionEvent event) {
        System.out.println("saveListAsClick");
    }

    public static void main(String[] args) {
        SwingUtilities.invokeLater(new Runnable() {
            public void run() {
                ProgramInfoWindow thisClass = new ProgramInfoWindow();
                thisClass.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
                thisClass.setVisible(true);
            }
        });
    }

}

File: converted_source/uprginfo.lfm.py
from common import *

#Useage:
#        import ProgramInfoWindow
#        window = ProgramInfoWindow.ProgramInfoWindow()
#        window.visible = 1

class ProgramInfoWindow(JFrame):
    def __init__(self, title='Program Information'):
        JFrame.__init__(self, title, windowClosing=self.OnClose)
        self.setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE) # HIDE_ON_CLOSE DO_NOTHING_ON_CLOSE
        self.BuildWindow()
        
    def OnClose(self, event):
        print "Closed"
        
    # a menu is top item on the menu bar and can be pulled down for a list of menu items
    def createMenu(self, name, mnemonic=None, menuBar = None):
        menu = JMenu(name)
        if mnemonic:
            menu.setMnemonic(mnemonic)
        if menuBar:
            menuBar.add(menu)
        return menu
        
    def BuildWindow(self):
        toolkit = java.awt.Toolkit.getDefaultToolkit()
        contentPane = self.getContentPane()
        contentPane.setLayout(None)
        # -- contentPane.setLayout(BoxLayout(contentPane, BoxLayout.Y_AXIS))
        buttonFont = Font("Arial Narrow", Font.PLAIN, 9)
        buttonMargin = Insets(1, 1, 1, 1)
        self.setBounds(310, 266, 450, 301  + 80) # extra for title bar and menu
        #  --------------- UNHANDLED ATTRIBUTE: self.TextHeight = 13
        #  --------------- UNHANDLED ATTRIBUTE: self.Scaled = False
        #  --------------- UNHANDLED ATTRIBUTE: self.OnResize = FormResize
        #  --------------- UNHANDLED ATTRIBUTE: self.PixelsPerInch = 96
        #  --------------- UNHANDLED ATTRIBUTE: self.AutoScroll = False
        #  --------------- UNHANDLED ATTRIBUTE: self.Position = poScreenCenter
        #  --------------- UNHANDLED ATTRIBUTE: self.OnCreate = FormCreate
        
        self.helpButtonImage = toolkit.createImage("../resources/ProgramInfoForm_helpButton.png")
        self.helpButton = SpeedButton('Help', ImageIcon(self.helpButtonImage), actionPerformed=self.helpButtonClick, font=buttonFont, margin=buttonMargin)
        self.helpButton.setMnemonic(KeyEvent.VK_H)
        self.helpButton.setBounds(332, 135, 113, 25)
        self.Close = JButton('Close', actionPerformed=self.CloseClick, font=buttonFont, margin=buttonMargin)
        self.Close.setMnemonic(KeyEvent.VK_C)
        self.Close.setBounds(332, 4, 113, 21)
        #  --------------- UNHANDLED ATTRIBUTE: self.Close.Cancel = True
        self.saveListAs = JButton('Save as...', actionPerformed=self.saveListAsClick, font=buttonFont, margin=buttonMargin)
        self.saveListAs.setMnemonic(KeyEvent.VK_S)
        self.saveListAs.setBounds(332, 83, 113, 21)
        self.copyToClipboard = JButton('Copy to Clipboard', actionPerformed=self.copyToClipboardClick, font=buttonFont, margin=buttonMargin)
        self.copyToClipboard.setMnemonic(KeyEvent.VK_P)
        self.copyToClipboard.setBounds(332, 55, 113, 21)
        #  --------------- UNHANDLED ATTRIBUTE: self.copyToClipboard.Cancel = True
        self.infoList = JTextArea(10, 10)
        self.infoList.setBounds(2, 2, 323, 295)
        #  --------------- UNHANDLED ATTRIBUTE: self.infoList.ScrollBars = ssVertical
        contentPane.add(self.helpButton)
        contentPane.add(self.Close)
        contentPane.add(self.saveListAs)
        contentPane.add(self.copyToClipboard)
        scroller = JScrollPane(self.infoList)
        scroller.setBounds(2, 2, 323, 295)
        contentPane.add(scroller)
        
    def CloseClick(self, event):
        print "CloseClick"
        
    def copyToClipboardClick(self, event):
        print "copyToClipboardClick"
        
    def helpButtonClick(self, event):
        print "helpButtonClick"
        
    def saveListAsClick(self, event):
        print "saveListAsClick"
        
if __name__ == "__main__":
    window = ProgramInfoWindow()
    window.visible = 1
    window.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE)
